features/support/api_helpers.rb

```diff
@@ -112,14 +112,42 @@ def register_interest_in_framework(framework_slug, supplier_id)
   end
 end
 
-def submit_supplier_declaration(framework_slug, supplier_id, declaration)
+def submit_supplier_declaration_and_sign_agreement(framework_slug, supplier_id, declaration)
   path = "/suppliers/#{supplier_id}/frameworks/#{framework_slug}/declaration"
   response = call_api(:put, path, payload: {
     declaration: declaration,
     updated_by: "functional tests",
   })
   [200, 201].should include(response.code), _error(response, "Failed to submit supplier declaration #{framework_slug} #{supplier_id}")
   JSON.parse(response.body)['declaration']
+  update_framework_agreement_status(framework_slug, supplier_id, true)
```

Comment:
Supplier frontend uses it to decide whether to show the 'View documents' and 'View your opportunities' links. Will do the renaming shortly.